File: previous/217.contains-duplicate/217.contains-duplicate.js
/*
 * @lc app=leetcode id=217 lang=javascript
 *
 * [217] Contains Duplicate
 * Time Complexity: O(n)
 * Space Complexity: O(n)
 * Notes: 與 1. Two Sum 類似, 都是使用 hash-map 解
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
	const obj = {};

	//build hash map
	for (let i = 0; i < nums.length; i++) {
		obj[nums[i]] = i;
	}

	//search on hash map
	for (let i = 0; i < nums.length; i++) {
		const value = nums[i];
		if (obj[value] != null && obj[value] !== i) return true;
	}

	return false;
};
// @lc code=end

File: previous/217.contains-duplicate/217.contains-duplicate.py
#
# @lc app=leetcode id=217 lang=python3
#
# [217] Contains Duplicate
#

# @lc code=start
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
      # sol 2 - set
      return len(nums) != len(set(nums))

# @lc code=end



# Sol 1 - dict count
# def containsDuplicate(self, nums: List[int]) -> bool:
#         map = {}
#         for n in nums:
#             map[n] = map.get(n, 0) + 1
#         for k, v in map.items():
#             if v > 1:
#                 return True
#         return False


